reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs

```diff
@@ -1,183 +1,238 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Text;
+using System.Linq;
 
-namespace SimpleCalculator.Business
+namespace SimpleCalculator.Business;
+
+public record Calculator
 {
-    public record Calculator
+    public static Dictionary<KeyInput, string> KeyTranslator = new()
+    {
+        { KeyInput.Multiplication, "×" },
+        { KeyInput.Division, "÷" },
+        { KeyInput.Subtraction, "−" },
+        { KeyInput.Addition, "+" },
+        { KeyInput.Back, "⌫" },
+        { KeyInput.Dot, "." },
+        { KeyInput.Clear, "C" },
+        { KeyInput.Equal, "=" },
+        { KeyInput.Percentage, "%" },
+        { KeyInput.PlusMinus, "±" },
+        { KeyInput.Zero, "0" },
+        { KeyInput.One, "1" },
+        { KeyInput.Two, "2" },
+        { KeyInput.Three, "3" },
+        { KeyInput.Four, "4" },
+        { KeyInput.Five, "5" },
+        { KeyInput.Six, "6" },
+        { KeyInput.Seven, "7" },
+        { KeyInput.Eight, "8" },
+        { KeyInput.Nine, "9" }
+    };
+
+    private string Number { get; init; }
+    private string Operator { get; init; }
+    private double? Number1 { get; init; }
+    private double? Number2 { get; init; }
+    private bool IsNumber2Percentage { get; init; }
+    private double? Result { get; init; }
+    private bool HasOperator => !string.IsNullOrEmpty(Operator);
+    private bool HasNumber => !string.IsNullOrEmpty(Number);
+    private bool HasNumber1 => Number1 != null;
+    private bool HasNumber2 => Number2 != null;
+
+    public string Output => $"{(Result != null ? Result.Value : HasNumber ? Number : "0")}";
+    public string Equation => $"{Number1} {Operator} {Number2}{(IsNumber2Percentage ? "%" : string.Empty)}{(Result != null ? $" =" : string.Empty)}";
+
+    public Calculator Input (KeyInput key)
+        => Input(this, key);
+
+    private Calculator Input (Calculator calculator, KeyInput key)
     {
-        private string Number { get; init; }
-        private string Operator { get; init; }
-        private double? Number1 { get; init; }
-        private double? Number2 { get; init; }
-        private bool IsNumber2Percentage { get; init; }
-        private double? Result { get; init; }
-        private bool HasOperator => !string.IsNullOrEmpty(Operator);
-        private bool HasNumber => !string.IsNullOrEmpty(Number);
-        private bool HasNumber1 => Number1 != null;
-        private bool HasNumber2 => Number2 != null;
-
-        public string Output => $"{(Result != null ? Result.Value : HasNumber ? Number : "0")}";
-        public string Equation => $"{Number1} {Operator} {Number2}{(IsNumber2Percentage ? "%" : string.Empty)}{(Result != null ? " =" : string.Empty)}";
-
-        public Calculator Input(string key)
-            => Input(this, key);
-
-        private Calculator Input(Calculator calculator, string key)
+        if (KeyTranslator.TryGetValue(key, out var keyValue))
```

Comment:
we already have a set value with the enum for the keys we shouldn't be trying to get a symbol here

Reply:
Changes in next commit